### PR TITLE
  Add periodic EOB completion check to Coordinator for bootstrap datastreams

### PR DESCRIPTION
**Pull request by mittalprince**
## Summary
- Added `periodicEobCheck` scheduled task in `Coordinator` that runs on the leader and checks whether all tasks for bootstrap datastreams have reached `COMPLETE` status
- Added `maybeMarkEOBCompleteForBootstrapDatastreams` which stamps `system.isEOBComplete=true` on all datastreams in the group (including deduped streams) when all tasks are complete
- Added `IS_EOB_COMPLETE` constant to `DatastreamMetadataConstants`
- Added `eobCheckPeriodMs` and `eobCheckConnectorNames` config to `CoordinatorConfig` with defaults (1 min period, all connectors)
- Added unit tests covering happy path, edge cases (no tasks, null status, update failure, multiple groups, connector allowlist, non-leader)

## Testing Done
- [x] Unit tests added for all edge cases in `TestCoordinator`
- [ ] Integration tests pass
- [ ] Manual testing performed

### CODE REVIEW

File: datastream-common/src/main/java/com/linkedin/datastream/common/DatastreamMetadataConstants.java [hunk 1]
```diff
@@ -143,4 +143,11 @@ public class DatastreamMetadataConstants {
    */
   public static final String CDC_BOOTSTRAP_REQUIRED_KEY = "system.cdcBootstrapRequired";
 
+  /**
+   * Set to "true" by the Coordinator when all tasks for a bootstrap datastream have reached
+   * {@link com.linkedin.datastream.server.DatastreamTaskStatus.Code#COMPLETE} status, indicating
+   * that End-of-Bootstrap (EOB) events have been successfully produced to all destination partitions.
+   */
+  public static final String IS_EOB_COMPLETE = "system.isEOBComplete";
```
Comment (kanishkjaiswal2015) [inline datastream-common/src/main/java/com/linkedin/datastream/common/DatastreamMetadataConstants.java:151]: should it be renamed as system.isBootstrapComplete?

File: datastream-server/src/main/java/com/linkedin/datastream/server/Coordinator.java [hunk 2]
```diff
@@ -1498,6 +1508,94 @@ && hasValidDestination(datastream) && !isDeletingOrExpired(datastream))
...
+    Map<String, List<DatastreamTask>> tasksByPrefix = assignmentByInstance.values()
+        .stream()
+        .flatMap(Set::stream)
+        .collect(Collectors.groupingBy(DatastreamTask::getTaskPrefix));
+
+    for (DatastreamGroup group : datastreamGroups) {
+      Datastream representative = group.getDatastreams().get(0);
+
+      // Bootstrap datastreams are identified by the "bst-" prefix in their name.
+      if (!representative.getName().contains("bst-")) {
```
Comment (kanishkjaiswal2015) [inline datastream-server/src/main/java/com/linkedin/datastream/server/Coordinator.java:1554]: this check can return false positives.
It would be better to check this via datastream's connector name.
